src/API/Endpoints/Reports/Endpoint.php

```diff
@@ -150,4 +150,43 @@ public function authorization_status_code() {
 		return $status;
 
 	}
+
+	/**
+	 * Get cached report
+	 *
+	 * @param WP_REST_Request $request Current request.
+	 */
+	public function get_cached_report( $request ) {
+
+		$query_args = [
+			'start' => $request['start'],
+			'end'   => $request['end'],
+		];
+
+		$cache_key = \Give_Cache::get_key( "api_get_report_{$this->endpoint}", $query_args );
+
+		$cached = \Give_Cache::get( $cache_key, false, $query_args );
+
+		return \Give_Cache::get( $cache_key, false, $query_args );
+	}
+
+	/**
+	 * Cache report
+	 *
+	 * @param WP_REST_Request $request Current request.
+	 */
+	public function cache_report( $request, $report ) {
+
+		$query_args = [
+			'start' => $request['start'],
+			'end'   => $request['end'],
+		];
+
+		$cache_key = \Give_Cache::get_key( "api_get_report_{$this->endpoint}", $query_args );
+
+		$result = \Give_Cache::set( $cache_key, $report, HOUR_IN_SECONDS, false, $query_args );
```

Comment:
@henryholtgeerts I will suggest you use `set_db_query` and `get_db_query` cache functions because it will auto flush cached query result when the form, donation or donor updates, you do not have to wait for 24 hours to get the fresh result.

Reply:
Thanks for the recommendation and walking me through the db_query logic! Changes made, ref: https://github.com/impress-org/givewp/pull/4437/commits/6aa9e106f1e35b31c3b2aa8c12a358ed251b96d0